src/Unleash/Variants/VariantUtils.cs

```diff
@@ -14,33 +14,35 @@ public static Variant SelectVariant(string groupId, UnleashContext context, List
         {
             var totalWeight = variantDefinitions.Sum(v => v.Weight);
 
-            if (totalWeight == 0) {
-                return null;
-            }
-
-            var variantOverride = GetOverride(variantDefinitions, context);
-            if (variantOverride != null)
+            if (totalWeight == 0)
             {
-                return variantOverride.ToVariant();
+                return null;
             }
 
-            var stickiness = variantDefinitions[0].Stickiness ?? "default";
+            var stickiness = variantDefinitions.FirstOrDefault()?.Stickiness ?? "default";
             var target = StrategyUtils.GetNormalizedNumber(GetIdentifier(context, stickiness), groupId, VARIANT_NORMALIZATION_SEED, totalWeight);
 
             var counter = 0;
+            Variant result = null;
             foreach (var variantDefinition in variantDefinitions)
             {
                 if (variantDefinition.Weight != 0)
                 {
+                    if (variantDefinition.Overrides.Count > 0 && variantDefinition.Overrides.Any(OverrideMatchesContext(context)))
+                    {
+                        result = variantDefinition.ToVariant();
```

Comment:
why not just return instead of breaking here?

Reply:
@gardleopard no real reason, I tend to try and keep the number of return statements from a method as low as possible but thats a 'me thing', theres no technical reason for it. I can change it if you'd like.